rust/lance/src/dataset.rs

```diff
@@ -833,13 +846,19 @@ impl Dataset {
     /// * `operation` - A description of the change to commit
     /// * `read_version` - The version of the dataset that this change is based on
     /// * `store_params` Parameters controlling object store access to the manifest
+    /// * `enable_v2_manifest_paths`: If set to true, and this is a new dataset, uses the new v2 manifest
+    ///   paths. These allow constant-time lookups for the latest manifest on object storage.
+    ///   This parameter has no effect on existing datasets. To migrate an existing
+    ///   dataset, use the [`Self::migrate_manifest_paths_v2`] method.
+    ///   Default is False.
```

Comment:
Why default to false?  Do we expect many people to beta test this feature?  Or is it because it doesn't apply for all object stores?

Reply:
Main reason this ain’t default is because older versions of Lance can’t read this scheme. So we should wait for a while before we make this default to reduce the likelihood users accidentally use old SDKs with this versioning scheme. Noted in this issue https://github.com/lancedb/lance/issues/2815